confmap/provider/eecprovider/config.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Dynatrace LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package eecprovider // import "github.com/Dynatrace/dynatrace-otel-collector/confmap/provider/eecprovider/eecprovider"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+)
+
+type config struct {
+	// A time duration that defines how frequently the provider should check the given URL for updates.
+	refreshInterval time.Duration
+
+	// A header that will be used to authenticate the provider with the EEC.
+	// We will always use Api-Token (and this can be assumed as a default if auth-file is passed),
+	// but upstream will want to support configurable headers and we should consider
+	// offering this option should upstream determine the option is required.
+	authHeader string
+
+	// Token passed as the value for the header in authHeader.
+	authToken string
+
+	// Whether to use HTTP instead of HTTPS
+	insecure bool
```

Comment:
Can we infer this from url scheme like `eec://` and `eecs://`? Maybe HTTP is getting in the way of my brain but it strikes me as a bit odd to include it as a query parameter.

Reply:
I suspect this option will be for local testing or edge cases, which is why I "hid" it in an option rather than offered it as a supported scheme. We can use the scheme, but I think for our specific use case it's a worse option.